upload_sig_storage_driver.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+VERSION=${1:?"Missing version number argument (arg 1)"}
+NEXUS_USER=${2:?"Missing Nexus username argument (arg 2)"}
+PRODUCT=${3:?"Missing product argument (arg 3)"}
+REGISTRY="registry.k8s.io/sig-storage"
+STACKABLE_REGISTRY="docker.stackable.tech/k8s/sig-storage"
+
+ARCHITECTURES=(
+    amd64
+    arm64
+)
+
+function pull_retag_and_push_images() {
+    for arch in "${ARCHITECTURES[@]}"; do
+        docker pull "${REGISTRY}/${PRODUCT}:v${VERSION}" --platform linux/${arch}
+        docker tag "${REGISTRY}/${PRODUCT}:v${VERSION}" "${STACKABLE_REGISTRY}/${PRODUCT}:v${VERSION}-${arch}"
+        docker push "${STACKABLE_REGISTRY}/${PRODUCT}:v${VERSION}-${arch}"
+    done
+}
+
+function create_and_push_manifest_list() {
+    docker manifest create "${STACKABLE_REGISTRY}/${PRODUCT}:v${VERSION}" --amend "${STACKABLE_REGISTRY}/${PRODUCT}:v${VERSION}-amd64" --amend "${STACKABLE_REGISTRY}/${PRODUCT}:v${VERSION}-arm64"
+    LIST_DIGEST=$(docker manifest push "${STACKABLE_REGISTRY}/${PRODUCT}:v${VERSION}")
+    cosign sign -y "${STACKABLE_REGISTRY}/${PRODUCT}@${LIST_DIGEST}"
+}
+
+function docker_login(){
+    read -r -s -p "Nexus Password: " NEXUS_PASSWORD
+    docker login -u $NEXUS_USER -p $NEXUS_PASSWORD docker.stackable.tech
+}
+
+function sign_images() {
+    for arch in "${ARCHITECTURES[@]}"; do
+        IMAGE_DIGEST=$(docker images --no-trunc --quiet "${STACKABLE_REGISTRY}/${PRODUCT}:v${VERSION}-${arch}")
+        cosign sign -y "${STACKABLE_REGISTRY}/${PRODUCT}@${IMAGE_DIGEST}"
+    done
+}
+
+main() {
```

Comment:
Just a question...all others have `function` in front of them. Is that optional in bash?

Reply:
Yes by the looks of this https://www.gnu.org/software/bash/manual/bash.html#Shell-Functions. 

We use the function keyword in our bash scripts everywhere else, I would just add this to main.

Reply:
Done https://github.com/stackabletech/docker-images/pull/668/commits/3082d37d8ce1bb1dd096628fb51e152c1823fffd